File: src/main/java/neo/idlib/containers/CFloat.java
package neo.idlib.containers;

// This class represents how floats are treated in C with pointers
public class CFloat {
    private float val = 0.0f;

    public CFloat() {
    }

    public CFloat(float val) {
        this.val = val;
    }

    public float getVal() {
        return val;
    }

    public void setVal(float val) {
        this.val = val;
    }


}

File: src/main/kotlin/neo/idlib/containers/CFloat.kt
package neo.idlib.containers

// This class represents how floats are treated in C with pointers
class CFloat {
    private var internalValue = 0.0f

    constructor()
    constructor(fromFloat: Float) {
        this.internalValue = fromFloat
    }

    fun getVal(): Float {
        return internalValue
    }

    fun setVal(fromFloat: Float) {
        this.internalValue = fromFloat
    }
}